spec/PeridicBackgroundSync-index.bs

```diff
@@ -0,0 +1,140 @@
+
+  As this API relies on service workers, functionality provided by this API is only available in a <a>secure context</a>.
+  <div class="example">
+    Requesting a <a>periodic Background Sync opportunity</a> at a daily frequency from a <a>browsing context</a>:
+
+    <pre class="lang-js">
+      function getLatestNews() {
+        const registration = await navigator.serviceWorker.ready;
+        try {
+          registration.periodicSync.register('get-latest-news', {
+            minInterval: 24 * 60 * 60 * 1000,
+          });
+        } catch (error) {
+          console.log('Periodic Sync could not be registered!');
+        }
+      }
+    </pre>
+
+    Reacting to a periodicsync event within a <a>service worker</a>:
+
+    <pre class="lang-js">
+      self.addEventListener('periodicsync', event => {
+        event.waitUntil(fetchAndCacheLatestNews());
+      });
+    </pre>
+
+    In the above example <code>fetchAndCacheLatestNews</code> is a developer-defined function.
+  </div>
+</section>
+
+<section>
+  <h2 id="concepts">Concepts</h2>
+
+  The periodicsync event is considered to run <dfn>in the background</dfn> if no <a>service worker clients</a> whose <a>frame type</a> is top-level or auxiliary exist for the origin of the corresponding service worker registration.
+
+  The user agent is considered to be <dfn>online</dfn> if the user agent has established a network connection. A user agent MAY use a stricter definition of being <a>online</a>. Such a stricter definition MAY take into account the particular <a>service worker</a> or origin a <a>periodicsync registration</a> is associated with.
+
+  A <dfn>periodic Background Sync opportunity</dfn> allows periodic synchronization between the server and the web app, the exact frequency of which is decided by the user agent. It can be requested via a <a>periodicsync registration</a>.
+</section>
+
+<section>
+  <h2 id="constructs">Constructs</h2>
+  A <a>service worker registration</a> has an associated <dfn>list of periodicsync registrations</dfn> whose element type is a <a>periodicsync registration</a>.
+  
+  A <dfn>periodicsync registration</dfn> is a tuple consisting of a <a>tag</a>, a dictionary of <a>options</a>, and a <a lt="registration state">state</a>.
+
+  The <dfn>tag</dfn> is a DOMString. Within one <a>list of periodicsync registrations</a> each <a>periodicsync registration</a> MUST have a unique <a>tag</a>.
+
+  The <dfn>options</dfn> are a dictionary containing <a>minInterval</a>, a long long. <dfn>minInterval</dfn> is used to specify the minimum interval, in milliseconds, at which the periodic synchronization should happen.
+
+  The <dfn>registration state</dfn> is one of <dfn>pending</dfn>, <dfn>waiting</dfn>, <dfn>firing</dfn>, or <dfn>reregisteredWhileFiring</dfn>. It is initially set to <a>pending</a>.
```

Comment:
I know this is forked from the BackgroundSync spec, but will this state be necessary for PBS?

Reply:
Yes, some algorithms I'll add will need them.